### PR TITLE
Extracted a package. Give a group flag to the authorized keys file

### PR DESCRIPTION
**Pull request by luis-falcon**
Made some changes discussed yesterday.

1) extracted into a package the functions and the main program in a main file. I added a group flag to the building of the authorization keys file to be able to create 1 authorized_keys file per organization (discussions with matt) and other. The behaviour is exactly the same as yesterday so far. feel free to comment and let me know if you think is a good idea to have those functionalities as a package. I thought it was because then we can create different programs per org or just one with a list of orgs and it seems easier that way. Please comment or pst me on slack.

### CODE REVIEW

File: user-authorization-builder/sresshd/sresshd.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+		Data: data,
+	}
+}
+
+//BuildConfigMapData builds the data to be put on the configMap from the authorized keys file
+func BuildConfigMapData(org string) (map[string]string, error) {
+
+	configMapData := make(map[string]string)
+
+	authFile := fmt.Sprintf("%s_authorized_keys", org)
```
Comment (mbarnes) [inline user-authorization-builder/sresshd/sresshd.go:135]: Could we leave the `org` out of the name and just use `authorized_keys`, so it's consistent across all the config maps.  I think that will make it easier for the controller to set up the volume mount.  I assume the org name will (eventually) be captured in the config map name.

File: user-authorization-builder/sresshd/sresshd.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+	return &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: annotations,
+			Name:        name,
+			Namespace:   namespace,
+		},
```
Comment (mbarnes) [inline user-authorization-builder/sresshd/sresshd.go:125]: We should add a label to the `ConfigMap` so the controller can find it.  I'm going to add a `LabelSelector` field to the CRD that specifies how to find these types of config maps.

Doesn't matter much what the label is for now... we can change it later.  But maybe just stub something in here?

**Comment by jharrington22** [comment]
@luis-falcon I would consider structuring this repo like other Go projects eg.

/cmd - main.go
/pkg -  packages
/hack  - scripts etc

**Comment by luis-falcon** [comment]
@jharrington22  I agree. However, this was to start and have the code other than my PC. We still do not know if we want to have this in a separate repository all together or just here as part of this repo an structured the way you suggested.

**Comment by jharrington22** [comment]
@luis-falcon np was just mentioning it since this is in the early stages.

**Comment by mbarnes** [comment]
Since this is still a work-in-progress...
/lgtm

**Pull request merged**